crates/biome_js_analyze/src/analyzers/nursery/no_unused_private_class_members.rs

```diff
@@ -0,0 +1,299 @@
+    ///  }
+    /// }
+    /// ```
+    ///
+    /// ```ts,expect_diagnostic
+    ///  class TsBioo {
+    ///    private unusedProperty = 5;
+    ///
+    ///    private unusedMethod() {
+
+    ///    };
+    ///  }
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// class UsedMember {
+    ///  #usedMember = 42;
+    ///
+    ///  method() {
+    ///		return this.#usedMember;
+    ///  }
+    /// }
+    /// ```
+    ///
+    pub(crate) NoUnusedPrivateClassMembers {
+        version: "next",
+        name: "noUnusedPrivateClassMembers",
+        recommended: false,
+        fix_kind: FixKind::Unsafe,
+    }
+}
+
+declare_node_union! {
+    pub(crate) AnyPrivateMember = AnyJsClassMember | TsPropertyParameter
+}
+
+#[derive(Debug, Clone)]
+pub struct RuleState {
+    range: TextRange,
+    member: AnyPrivateMember,
+}
+
+impl Rule for NoUnusedPrivateClassMembers {
+    type Query = Ast<JsClassDeclaration>;
+    type State = RuleState;
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let private_members: FxHashMap<TokenText, RuleState> =
+            get_all_declarated_private_members(node);
+
+        if private_members.is_empty() {
+            return vec![];
+        }
+
+        traverse_members_usage(node.syntax(), private_members)
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        Some(RuleDiagnostic::new(
+            rule_category!(),
+            state.range,
+            markup! {
+                "This private class member is defined but never used."
+            },
+        ))
+    }
+
+    fn action(ctx: &RuleContext<Self>, state: &Self::State) -> Option<JsRuleAction> {
+        let mut mutation = ctx.root().begin();
+
+        mutation.remove_node(state.member.clone());
+
+        Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Remove unused declaration." }.to_owned(),
+            mutation,
+        })
+    }
+}
+
+/// Check for private member usage
+/// if the member usage is found, we remove it from the hashmap
+fn traverse_members_usage(
+    syntax: &SyntaxNode<JsLanguage>,
+    mut private_members: FxHashMap<TokenText, RuleState>,
+) -> Vec<RuleState> {
+    let iter = syntax.preorder();
+
+    for walk in iter {
+        match walk {
+            biome_rowan::WalkEvent::Enter(node) => {
+                if let Some(js_name) = AnyJsName::cast_ref(&node) {
+                    let value_token = match &js_name {
+                        AnyJsName::JsName(name) => name.value_token(),
+                        AnyJsName::JsPrivateName(name) => name.value_token(),
+                    };
+
+                    if let Ok(value_token) = value_token {
+                        let token = value_token.token_text_trimmed();
```

Comment:
We could just use `text_trimmed`?
```suggestion
                        let name = value_token.text_trimmed();
```

`TokenText` is useful when you wish own the string representation of a token without allocating a string. If we just need `&str`, then we can directly call `text_trimmed()`.

Reply:
In that case the token is just needed to get the state from private_members hash, If i use text_trimmed() i'll need to update the hash signature to <&str, RuleState>, right?

Reply:
Oh I see. Thus, don't take my comment into consideration.